test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -206,54 +223,86 @@ DEFINE_TEST_1(test_inclusive_scan_by_segment, BinaryOperation)
     }
 };
 
-template<typename _Tp>
+template <typename _Tp>
+struct UserBinaryPredicate
+{
+    bool
+    operator()(const _Tp& __x, const _Tp& __y) const
+    {
+        using KeyT = ::std::decay_t<decltype(__y)>;
+        return __y != KeyT(1);
```

Comment:
May we use ```1```int the right part of comparison instead of ```KeyT(1)``` ?

Reply:
This is okay for primitive datatypes but not okay for types such as `std::complex` where 1 cannot be converted to `std::complex`. For this reason it should be left as is.

Reply:
ok